src/StoreKeeper/WooCommerce/B2C/Models/AbstractModel.php

```diff
@@ -341,4 +342,19 @@ public static function purge(): int
     {
         return 0;
     }
+
+    /**
+     * @throws Exception
+     */
+    protected static function getValidForeignFieldKey(string $attributeForeignKey, string $tableName): string
+    {
+        /* @since 9.0.8 */
+        if (strlen($attributeForeignKey) > static::MAX_FOREIGN_KEY_LENGTH) {
+            $allowedForeignKeyLength = static::MAX_FOREIGN_KEY_LENGTH - strlen($tableName) - 4; // -4 for underscores and fk prefix
+            $attributeTableForeignKeyHash = substr(bin2hex(random_bytes($allowedForeignKeyLength)), 0, $allowedForeignKeyLength);
+            $attributeForeignKey = "{$tableName}_{$attributeTableForeignKeyHash}_fk";
+        }
```

Comment:
```suggestion
        if (strlen($attributeForeignKey) > static::MAX_FOREIGN_KEY_LENGTH) {
            $attributeTableForeignKeyHash = hash('crc32', $data, false);
            $attributeForeignKey = "{$tableName}_{$attributeTableForeignKeyHash}_fk";
        }
```
I think we need to make it stable not random, otherwise it it would be really hard to remove the key after, because you don't know what it will it be

Reply:
Used crc32 as hash.